### PR TITLE
[DTM] SOFT-4083 [11/23]: Shape-tolerant preset values

### PR DESCRIPTION
**Pull request by pauloiankoski**
Resolves responsive-envelope and per-corner preset values instead of stringifying them, and preserves untouched non-scalar properties on save rather than flattening them.

## Test instructions

1. Run `npx wp-scripts test-unit-js src/style-library/__tests__/presets.test.js`.
2. The important check is the save path: open a preset, change **one** field, save, then inspect the stored document and confirm the properties you did not touch kept their original shape rather than being flattened to a string.
3. This is what stops a richer value written by another ticket from being destroyed by an unrelated save here.

<details>
<summary>Setup (same for every slice in this stack)</summary>

```bash
git checkout SOFT-4083/phase-6a-shape-tolerant-reads
bun run build-wp
```

The Style Library is at **WP Admin → Kadence → Style Library**. The Button preset screen is under **Block Presets → Button**. Block-editor checks use a **Kadence Single Button** block on any post.
</details>

---

Slice 11 of 23 in the SOFT-4083 stack. Read bottom-up: this PR is understandable from its own diff plus the slice below it.

## Summary by CodeRabbit

* **Bug Fixes**
  * Preset saves now preserve responsive and per-corner values when unrelated settings are edited.
  * Improved handling of non-scalar token values during preset updates.
  * Unsupported token formats are handled safely without corrupting saved data.

* **Tests**
  * Added coverage for responsive, per-corner, invalid, and unchanged preset values.
  * Added coverage for creating and updating presets while preserving stored token structures.

### CODE REVIEW

**Review by d4mation — APPROVED** [review]

File: src/style-library/helpers/presets.js [hunk 1]
```diff
@@ -110,20 +129,82 @@ export function idToAlias(value) {
...
+ *
+ * @param {*} value The stored token entry.
+ *
+ * @since TBD
+ *
+ * @return {boolean} True when `value` is a per-corner slot list.
+ */
+function isSlotList(value) {
+	return Array.isArray(value) && value.length === SLOT_LIST_SIDES;
+}
```
[the remark this answers is not in the document]
Comment (pauloiankoski) [reply src/style-library/helpers/presets.js:160]: Took the resolver half: any unresolvable corner now returns `""` instead of a partial shorthand. Left `isSlotList` broad on purpose — it also gates the read-only field, and narrowing it would make a malformed slot list look editable. Tests added.

**Review by d4mation — APPROVED** [review]
